src/dbnode/storage/bootstrap.go

```diff
@@ -70,42 +70,154 @@ const (
 
 type bootstrapFn func() error
 
-type bootstrapManager struct {
-	sync.RWMutex
-
-	database                    database
-	mediator                    databaseMediator
+type instrumentation struct {
 	opts                        Options
 	log                         *zap.Logger
-	bootstrapFn                 bootstrapFn
 	nowFn                       clock.NowFn
 	sleepFn                     sleepFn
-	processProvider             bootstrap.ProcessProvider
-	state                       BootstrapState
-	hasPending                  bool
 	status                      tally.Gauge
 	bootstrapDuration           tally.Timer
+	bootstrapNamespacesDuration tally.Timer
 	durableStatus               tally.Gauge
 	lastBootstrapCompletionTime xtime.UnixNano
+	start                       time.Time
+	startNamespaces             time.Time
+	logFields                   []zapcore.Field
```

Comment:
I believe moving these variables here (which were local to a function/method) might introduce thread safety issues, doesn't it? I had a brief look at the code and it's not immediately clear if it's true or not. Either way, I think it might be better keep them local than to spend time considering thread safety.

Reply:
I don't think bootstrap is called concurrently. Maybe I am wrong?  I could try to make instrumentation methods thread-safe but I'm not sure how it would look like now, I guess if readability isn't affected then it's worth having thread-safe implementation.

Reply:
The only relevant field from the original implementation that was accessed with a mutex was `lastBootstrapCompletionTime` (I guess for exposing it to concurrent reads), and I've suggested not to move it to `instrumentation`.
Metrics objects are thread safe by design (so that their values can be scraped asynchronously), so they are not an ussue.